src/types/composite/container.ts

```diff
@@ -28,7 +26,7 @@ export class ContainerType<T extends ObjectLike = ObjectLike> extends CompositeT
     this._typeSymbols.add(CONTAINER_TYPE);
   }
   isVariableSize(): boolean {
-    return Object.values(this.fields).some((fieldType) => fieldType.isVariableSize());
+    return Object.values(this.fields as Record<string, Type<unknown>>).some((fieldType) => fieldType.isVariableSize());
```

Comment:
Is this type coersion needed?

Reply:
unfortunately yes. Because interfaces don't have index signature, we have to add `{}` which casuses fields to be Type | never . Never is the case if there is no key in given object. I have no idea how to fix that like we cannot create generic that accepts every interface unless we relax it with `{}` which basically allows anything